test/tools/ossfuzz/protomutators/YulProtoMutator.cpp

```diff
@@ -0,0 +1,2081 @@
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				for (auto &stmt: *_message->mutable_statements())
+					if (stmt.has_ifstmt())
+					{
+						delete stmt.release_ifstmt();
+						stmt.clear_ifstmt();
+						break;
+					}
+			},
+			_message,
+			_seed,
+			YPM::s_lowIP,
+			"Remove if statement from statement block"
+		);
+	}
+);
+
+// Add switch statement
+static YPR<Block> addSwitchStmt(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator& _rand)
+			{
+				auto switchStmt = new SwitchStmt();
+				switchStmt->add_case_stmt();
+				Expression *switchExpr = new Expression();
+				switchExpr->set_allocated_varref(YPM::varRef(_rand()));
+				switchStmt->set_allocated_switch_expr(switchExpr);
+				_message->add_statements()->set_allocated_switchstmt(switchStmt);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Add switch statement to statement block"
+		);
+	}
+);
+
+// Remove switch statement
+static YPR<Block> removeSwitchStmt(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				for (auto &stmt: *_message->mutable_statements())
+					if (stmt.has_switchstmt())
+					{
+						delete stmt.release_switchstmt();
+						stmt.clear_switchstmt();
+						break;
+					}
+			},
+			_message,
+			_seed,
+			YPM::s_lowIP,
+			"Remove switch statement from statement block"
+		);
+	}
+);
+
+// Add function call to statement block
+static YPR<Block> addFuncCall(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator& _rand)
+			{
+				auto call = new FunctionCall();
+				YPM::configureCall(call, _rand);
+				_message->add_statements()->set_allocated_functioncall(call);
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Add function call to statement block"
+		);
+	}
+);
+
+// Remove function call
+static YPR<Block> removeFuncCall(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				for (auto &stmt: *_message->mutable_statements())
+					if (stmt.has_functioncall())
+					{
+						delete stmt.release_functioncall();
+						stmt.clear_functioncall();
+						break;
+					}
+			},
+			_message,
+			_seed,
+			YPM::s_highIP,
+			"Remove function call from statement block"
+		);
+	}
+);
+
+// Add variable declaration
+static YPR<Block> addVarDecl(
+	[](Block* _message, unsigned int _seed)
+	{
+		YPM::functionWrapper<Block>(
+			[](Block* _message, YulRandomNumGenerator&)
+			{
+				_message->add_statements()->set_allocated_decl(new VarDecl());
+				// Hoist var decl to beginning of block
+				if (_message->statements_size() > 1)
+					_message->mutable_statements(0)->Swap(
+						_message->mutable_statements(_message->statements_size() - 1)
```

Comment:
What if the first statement was a var decl itself? Couldn't the code potentially become invalid?